src/app.rs

```diff
@@ -289,6 +291,19 @@ where
             states.stop(StateData::new(world, &mut self.data));
         }
 
+        // Read the Trans queue and apply changes.
+        {
+            let trans = {
```

Comment:
why drain and collect and then `into_iter()` later though? why not just grab the `Write<VecDeque<_>>` and `into_iter()` directly?

Reply:
Because I need to empty the queue and read the collected elements. If it wasn't from the borrow issues, I would have write_storage().drain(..).into_iter()